module/dkg/broker.go

```diff
@@ -127,13 +146,19 @@ func (b *Broker) Broadcast(data []byte) {
 		if err != nil {
 			b.log.Fatal().Err(err).Msg("create retry mechanism")
 		}
-		maxedExpRetry := retry.WithMaxRetries(retryMax, expRetry)
+		maxedExpRetry := retry.WithMaxRetries(retryMaxPublish, expRetry)
 
 		attempts := 1
 		err = retry.Do(context.Background(), maxedExpRetry, func(ctx context.Context) error {
-			err := b.dkgContractClient.Broadcast(bcastMsg)
+			err := b.dkgContractClient().Broadcast(bcastMsg)
 			if err != nil {
 				b.log.Error().Err(err).Msgf("error broadcasting, retrying (%x)", attempts)
+
+				// retry with next fallback client after 2 failed attempts
+				if attempts%2 == 0 {
+					b.log.Warn().Msgf("broadcast: retrying on attempt (%x) with fallback access node", attempts)
+					b.updateActiveDKGContractClient()
```

Comment:
Here I would actually call the update first, and then print the `activeQCContractClient` in the message, so that the engineer trying to debug this from logs will know which node is attempted when.

Reply:
https://github.com/onflow/flow-go/pull/1395/commits/d8c099575b498fb6e219121d0bf2a0910a40217a